h/browser/chrome/lib/hypothesis-chrome-extension.js

```diff
@@ -169,15 +180,33 @@ function HypothesisChromeExtension(dependencies) {
       return Promise.resolve();
     }
 
-    if (state.isTabActive(tab.id)) {
-      return sidebar.injectIntoTab(tab).catch(function (err) {
+    function inject(tab) {
+      sidebar.injectIntoTab(tab).catch(function (err) {
         tabErrors.setTabError(tab.id, err);
         state.errorTab(tab.id);
       });
     }
-    else if (state.isTabInactive(tab.id)) {
+
+    if (state.isTabActive(tab.id)) {
+      return uriinfo(tab.url).then(function(info) {
+        if (info.blocked) {
+            tabErrors.setTabError(
+              tab.id, new errors.BlockedSiteError(
+                "Hypothesis doesn't work on this site yet."));
+            state.errorTab(tab.id);
+        } else {
+          inject(tab);
+        }
+      }).catch(function() {
```

Comment:
It might be clearer to use the `.then(success, failure)` form here. `.then(success).catch(failure)` is always a bit weird because if you return a promise from `success` then you'll start catching things you didn't intend to catch. No big deal, though.